src/common/prop.c

```diff
@@ -162,44 +163,89 @@ daos_prop_merge(daos_prop_t *old_prop, daos_prop_t *new_prop)
 }
 
 static bool
-daos_prop_str_valid(d_string_t str, const char *prop_name, size_t max_len)
+daos_prop_str_format_valid(const char *str, const char *regex)
+{
+	regex_t		regx;
+	int		rc;
+
+	rc = regcomp(&regx, regex, REG_EXTENDED | REG_ICASE);
```

Comment:
Is it worth checking for `regex == NULL` here? Or do we just trust that `regcomp()` does the right thing? Not sure what the best practice is.

Reply:
I can add an assert for each of the arguments. Intention is the internal function will always be called with non-NULL arguments, e.g., as is done by the (current code) single caller daos_prop_str_valid(). Adding asserts can defend against future code changes that might mistakenly pass NULL args.

Reply:
I'm not against REG_ICASE, but just want to confirm this is intentional and indeed the rule for our property values (I think the keys are case sensitive)...?